File: 868-PushDominoes/868-PushDominoes.py
# Last updated: 6/11/2026, 11:30:54 AM
class Solution:
    def pushDominoes(self, dominoes: str) -> str:
        n = len(dominoes)
        force = [0]*n

        # left to right
        f = 0
        for i in range(len(dominoes)):
            if(dominoes[i] == "L"):
                f = 0
            elif(dominoes[i] == "R"):
                f = n
            else:
                f = max(f-1, 0)
            force[i] = f
        
        # right to left
        f = 0
        for i in range(n-1, -1, -1):
            if(dominoes[i] == "R"):
                f = 0
            elif(dominoes[i] == "L"):
                f = n
            else:
                f = max(f-1, 0)
            force[i] -= f
        
        print(force)
        ans = ""
        for i in force:
            if i>=1:
                ans+="R"
            elif i<=-1:
                ans += "L"
            else:
                ans += "."
        return ans

File: 868-PushDominoes/868-PushDominoes.cpp
// Last updated: 6/11/2026, 11:30:50 AM
class Solution {
public:
    string pushDominoes(string s) {
        int i = 0;
        char prv = '.';
        while(i<s.size()){
            if(s[i] == '.'){
                int j = i;
                while(j<s.size() && s[j] == '.'){
                    j++;
                }
                if(j == s.size()){
                    if(prv == 'L' || prv  == '.'){
                        i = j;
                    }else{
                        j = i;
                        while(j<s.size() && s[j] == '.'){
                            s[j] = 'R';
                            j++;
                        }
                        i = j;
                    }
                }else{
                    if(s[j] == 'R'){
                        if(prv == 'R'){
                            int cur = i;
                            while(cur<j){
                                s[cur] = 'R';
                                cur++;
                            }
                            i = j;
                        }else{
                            i = j;
                        }
                    }else{
                        if(prv == 'L' || prv == '.'){
                            int cur = j;
                            while(cur>=i){
                                s[cur] = 'L';
                                cur--;
                            }
                            i = j;
                        }else{
                            int l = i;
                            int r = j-1;
                            while(l<r){
                                s[l] = 'R';
                                s[r] = 'L';
                                l++;
                                r--;
                            }
                            i = j;
                        }
                    }
                }
            }else{
                prv = s[i];
                i++;
            }
        }

        return s;
    }
};